components/sharing-dialog/src/helpers/helpers.js

```diff
@@ -0,0 +1,249 @@
+import {
+    ACCESS_VIEW_ONLY,
+    ACCESS_NONE,
+    ACCESS_VIEW_AND_EDIT,
+    SHARE_TARGET_EXTERNAL,
+    SHARE_TARGET_PUBLIC,
+} from '../sharing-constants.js'
+
+/**
+ * Returns a function, that, as long as it continues to be invoked, will not be triggered. The
+ * function will be called after it stops being called for N milliseconds. If `immediate` is
+ * passed, trigger the function on the leading edge, instead of the trailing.
+ */
+
+export const debounce = (func, wait, immediate) => {
+    let timeout
+
+    return (...args) => {
+        const context = this
+
+        const later = () => {
+            timeout = null
+
+            if (!immediate) {
+                func.apply(context, args)
+            }
+        }
+
+        const callNow = immediate && !timeout
+
+        clearTimeout(timeout)
+        timeout = setTimeout(later, wait)
+
+        if (callNow) {
+            func.apply(context, args)
+        }
+    }
+}
+
+/**
+ * Name conversion
+ */
+
+export const nameToInitials = (name) => {
+    if (!name) {
+        return ''
+    }
+
+    const parts = name.split(' ')
+    const firstNameInitial = parts.length > 0 ? parts.shift().charAt(0) : ''
+    const lastNameInitial = parts.length > 0 ? parts.pop().charAt(0) : ''
+
+    return `${firstNameInitial}${lastNameInitial}`
```

Comment:
Are middle names being ignored on purpose? I think with western names this might be fine, but I wouldn't bet on every culture doing it the same way (even if they use the Latin alphabet)

Reply:
Yeah good question. This area of the code has now been replaced by the user-avatar, but I think that that does the same: https://github.com/dhis2/ui/blob/master/components/user-avatar/src/text-avatar.js. Probably good to find a less western-name-specific solution.

Reply:
Added it to the follow ups.